posthog/models/property/util.py

```diff
@@ -593,6 +599,9 @@ def get_property_string_expr(
     if allow_denormalized_props and (property_name, "properties") in materialized_columns:
         return f'{table_string}"{materialized_columns[(property_name, "properties")]}"', True
 
+    if allow_denormalized_props and (property_name, "person_properties") in materialized_columns:
```

Comment:
Hmm, I don't think this would work like we want it to.

Consider the case where we want the `email` person property, and both `email` on events & persons has been materialised.

In this case, based on ordering of these, we can easily return the wrong column.

So, I think we need to make it explicit which column we want here. Which in turn means passing this in as a parameter & updating everywhere.

Will write a test case for this as well^

Reply:
Yep, this test was failing & have updated all places to request the table column themselves, thus making this deterministic & immune to failures.